tests/cupy_tests/lib_tests/test_polynomial.py

```diff
@@ -0,0 +1,116 @@
+import unittest
+
+import pytest
+import numpy
+
+import cupy
+from cupy import testing
+
+
+@testing.gpu
+class TestPoly1d(unittest.TestCase):
+
+    def make_poly1d(self, xp, dtype):
+        a1 = testing.shaped_arange((4,), xp, dtype)
+        a2 = xp.zeros((4,), dtype)
+        b1 = xp.poly1d(a1)
+        b2 = xp.poly1d(a2)
+        return b1, b2
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_poly1d(self, xp, dtype):
+        a = testing.shaped_arange((5,), xp, dtype)
+        return xp.poly1d(a).coeffs
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_poly1d_leading_zeros(self, xp, dtype):
+        a = xp.array([0, 0, 1, 2, 3], dtype)
+        return xp.poly1d(a).coeffs
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_equal()
+    def test_poly1d_variable(self, xp, dtype):
+        a = testing.shaped_arange((5,), xp, dtype)
+        return xp.poly1d(a, variable='p').variable
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_equal()
+    def test_poly1d_order(self, xp, dtype):
+        a = testing.shaped_arange((10,), xp, dtype)
+        return xp.poly1d(a).order
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_equal()
+    def test_poly1d_eq(self, xp, dtype):
+        a, b = self.make_poly1d(xp, dtype)
```

Comment:
make_poly1d generates only `a != b` arrays.

Reply:
Yes, both numpy and cupy will return False in this case

Reply:
Please check also for `a == b` case.

Reply:
Please address this comment also in __ne__ test.